game-core/src/main/java/games/strategy/engine/framework/GameRunner.java

```diff
@@ -393,4 +413,14 @@ public static void clientLeftGame() {
   public static void quitGame() {
     mainFrame.dispatchEvent(new WindowEvent(mainFrame, WindowEvent.WINDOW_CLOSING));
   }
+
+  /**
+   * Execution context for {@link GameRunner}.
+   */
+  @AllArgsConstructor(access = AccessLevel.PRIVATE)
+  @Builder
+  public static final class Context {
+    public final String[] args;
+    public final Class<?> mainClass;
```

Comment:
This seems a little bit weird to me, I can imagine a couple of explanations for this type of pattern, but I'd like to hear your reasoning, why this extra field was introduced.

Reply:
You're talking about passing a `Context` to `GameRunner`, right?  It's simply a form of dependency injection.  If your question is why am I passing a _Parameter Object_ to `start()` instead of the two parameters individually, please see my answer to your next question below.

Or are you talking about why `mainClass` was introduced?  I was hoping that would be obvious. :smile:

Reply:
Latter question:
I was assuming that it's being used to avoid the "cyclic dependency" from HeadedRunner and back.
But other than that I'm not really getting behind it.

Reply:
Well, it's kind of required.  `GameRunner` can't refer to `HeadedGameRunner` because there's no dependency from `game-core` to `game-headed`.  Nor should there be, otherwise it kind of makes the whole modularization exercise moot.  Even if you wanted to introduce a circular dependency between `game-core` and `game-headed`, Gradle won't permit this.

I guess you could hard-code the `HeadedGameRunner` class name as a `String` within `GameRunner`, but that seems like another detail that could be easily overlooked if we perform a rename or package move in the future.

Reply:
Oh right, I was already one step too far, I thought for whatever reason that GameRunner was part of the game-headed project, simply because it should be in the end.

Reply:
Yes.  I actually foresee the _code_ in `GameRunner` gradually migrating to `HeadedGameRunner`, so that, in the end, `GameRunner` will no longer exist.